torchacc/dist/distributed_parallel.py

```diff
@@ -71,3 +72,32 @@ def forward_backward(self, *args, output_fn=None, **kwargs):
                 "forward_backward is only supported for pipeline parallel.")
         assert isinstance(self.model, PipelineParallel)
         return self.model.forward_backward(*args, output_fn=output_fn, **kwargs)
+
+    def sharded_optim_state_dict(self, optim: torch.optim.Optimizer):
+        if not self.has_fsdp:
+            raise NotImplementedError(
+                "sharded_optim_state_dict is only support for FullyShardedDataParallel"
+            )
+        assert isinstance(self.model, FullyShardedDataParallel)
```

Comment:
better to check not `self.spmd_fsdp` here and raise error to remind the user

Reply:
done